File: Python/프로그래머스/lv3/12927. 야근 지수/야근 지수.py
import heapq
def solution(n, works):
    answer = 0
    heap = []
    
    for work in works:
        heapq.heappush(heap,-work)
    while n != 0:
        n -= 1
        heapq.heappush(heap,heapq.heappop(heap)+1)
    
    return sum([i**2 for i in heap if i <= 0])

File: Python/프로그래머스/lv3/12927. 야근 지수/야근 지수.java
import java.util.PriorityQueue;
import java.util.Collections;

class Solution {
    public long solution(int n, int[] works) {
        long answer = 0;
        PriorityQueue<Integer> queue = new PriorityQueue<>(Collections.reverseOrder());
        for (int work : works) {
            queue.add(work);
        }
        while(n>0){
            n--;
            queue.add(queue.poll() - 1);
        }
        for (Integer i : queue) {
            if(i>0) answer += Math.pow(i,2);
        }
        return answer;
    }
}